linux/guest_customization/linux_gosc_start.yml

```diff
@@ -72,14 +78,26 @@
     gosc_cloudinit_log_file: "/var/log/cloud-init.log"
 
 # Wait GOSC completes message in logs
-- include_tasks: wait_gosc_complete_msg.yml
+- name: "Wait for traditional GOSC complete message"
+  include_tasks: wait_gosc_complete_msg.yml
   vars:
     wait_gosc_log_file: "{{ gosc_deploypkg_log_file }}"
     wait_gosc_msg_regexp: "Ran DeployPkg_DeployPackageFromFile successfully"
   when: not enable_cloudinit_gosc | bool
 
-- include_tasks: wait_gosc_complete_msg.yml
-  vars:
-    wait_gosc_log_file: "{{ gosc_cloudinit_log_file }}"
-    wait_gosc_msg_regexp: "Cloud-init .*finished at.*Datasource DataSourceOVF \\[seed=vmware-tools\\]"
+- name: "Wait for cloud-init GOSC complete message"
+  block:
+    - name: "Wait for cloud-init GOSC complete message when cloud-init version is lower than 23.1"
+      include_tasks: wait_gosc_complete_msg.yml
+      vars:
+        wait_gosc_log_file: "{{ gosc_cloudinit_log_file }}"
+        wait_gosc_msg_regexp: "Cloud-init .*finished at.*Datasource DataSourceOVF \\[seed=vmware-tools\\]"
```

Comment:
How about combining the two tasks into one by just setting the variable wait_gosc_msg_regexp different values before the task? 
Of course, not for the logic consideration, just wonder if there's any performence difference between the two approaches. If no, keep it as it is. I'm Ok with it.

Reply:
Thanks for your suggestion. But I prefer this way as it is more clear. I think you raised a good question about performance. I will appreciate that if you can propose any performance improvement methods.